src/main/java/com/microsoft/azure/vmagent/AzureVMAgent.java

```diff
@@ -555,27 +553,10 @@ public synchronized void deprovision(Localizable reason) throws Exception {
 
         if ((launcher instanceof AzureVMAgentSSHLauncher)) {
             AzureVMAgentSSHLauncher azureLauncher = (AzureVMAgentSSHLauncher) launcher;
-
-            OutputStream terminateResults = new OutputStream() {
-                private String terminateResultBuffer = "";
-
-                @Override
-                public void write(int b) throws IOException {
-                    terminateResultBuffer += (char) b;
-                }
-
-                @Override
-                public void flush() {
-                    if (!terminateResultBuffer.isEmpty()) {
-                        LOGGER.log(Level.FINER, "AzureVMAgent: deprovision: terminateScript, " + terminateResultBuffer);
-                        terminateResultBuffer = "";
-                    }
-                }
-            };
-
-            PrintStream terminateStream = new PrintStream(terminateResults, true, "UTF-8");
+            PrintStream terminateStream = new LogTaskListener(LOGGER, Level.INFO).getLogger();
 
             final boolean isUnix = this.getOsType().equals(OperatingSystemTypes.LINUX);
+            boolean skipTerminateScript = StringUtils.isNotBlank(terminateScript);
```

Comment:
Use `StringUtils.isBlank`?

Reply:
Oh good catch, I think I used the wrong one.